beacon-chain/sync/service.go

```diff
@@ -47,14 +50,35 @@ type Config struct {
 	CrystallizedStateBufferSize     int
 }
 
+// FinalizedBlock stores information regarding a block, its finalized epoch and the contents
+// of its crystallized state.
+type FinalizedBlock struct {
+	BeaconBlock        *types.Block
+	LastFinalizedEpoch uint64
+	CrystallizedState  *types.CrystallizedState
+}
+
 // DefaultConfig provides the default configuration for a sync service.
 func DefaultConfig() Config {
 	return Config{100, 100, 100, 100, 100, 100}
 }
 
 // NewSyncService accepts a context and returns a new Service.
 func NewSyncService(ctx context.Context, cfg Config, beaconp2p types.P2P, cs types.ChainService) *Service {
+	// 1 represents default sync and 0 , initial sync
+	mode := 1
 	ctx, cancel := context.WithCancel(ctx)
+	stored, err := cs.HasStoredState()
+
+	if err != nil {
+		log.Errorf("error retrieving stored state: %v", err)
+		cancel()
+		return nil
```

Comment:
if you fail to retrieve stored state, shouldn't you keep going and re-sync from the beginning?

Reply:
Yeah it should , will get it changed